src/components/QrcodeStream.vue

```diff
@@ -425,4 +462,19 @@ const videoElStyle = computed<CSSProperties>(() => {
     }
   }
 })
+
+// Component specific types
+
+/**
+ * Defines interface for object returned in track propery / callback which is "enriched" barcode scan result.
+ */
+interface AdjustedCode {
+  cornerPoints: {
+    x: number
+    y: number
+  }[]
+  boundingBox: DOMRectReadOnly
+  rawValue: string
+  format: DetectedBarcode['format']
+}
```

Comment:
I think the original type `DetectedBarcode` from `barcode-detector` should also work. It has the same structure. What's "adjusted" about it is just that all coordinates are re-calculated. Initially all coordinates are relative to the resolution of the camera but we have to scale/translate/crop that to fit into the area covered by the UI component. Accordingly, we "adjust" the coordinates of the detected barcodes, so what the tracking function paints is in the right position.

Reply:
That was not obvious :)

`cornerPoints.map((point) => translate(scale(point)))` evaluates to `Point2D[]` while expected type is a tuple `[ Point2D,  Point2D, Point2D, Point2D]`.